docs/contributing/design-documents.rst

```diff
@@ -0,0 +1,74 @@
+:title: Design Documents
+:description: Information necessary for a feature design document.
+
+.. _design-documents:
+
+Design Documents
+================
+
+Before submitting a pull request which is expected to significantly change the
+behavior of Deis (such as when contributing a new feature or refactor),
+contributors should first open an issue representing a design document.
+
+Goals
+-----
+
+Design documents were adopted in order to:
+
+* Involve stakeholders as early as possible in a feature's development
+* Ensure code changes accomplish the original motivations and design goals
+* Establish clear acceptance criteria for a feature or change
+* Enforce test-driven design methodology and automated test coverage
+
+Contents
+--------
+
+Design document issues should be named ``Design Doc: <change description>`` and
+contain the following sections.
+
+Goal
+^^^^
+
+This section should briefly describe the proposed change and the motivations
+behind the change. Tests will be written to ensure this design goal is met by
+the change.
+
+This section will typically reference a separate GitHub issue tracking
+the feature or change, which will typically be assigned to a release milestone.
+
+Code Changes
+^^^^^^^^^^^^
+
+This section should detail the code changes necessary to accomplish the change,
+as well as the proposed implementation. This should be as detailed as necessary to
+help reviewers understand the change.
+
+Tests
+^^^^^
+
+All changes should be covered by automated tests, either unit or integration tests
+(ideally both). This section should detail how tests will be written to validate
+that the change accomplishes the design goals and doesn't introduce any regressions.
+
+If a change cannot be sufficiently covered by automated testing, the design
+should be reconsidered. If there is no test coverage whatsoever for an affected
+section of code, a separate issue should be filed to integrate automated testing
+with that section of the codebase.
```

Comment:
Yes--if it's unreasonable to write a test for the feature because there isn't any example already of how to do so, that's a high-priority issue for Deis now that we must fix now.